src/openvic-simulation/Modifier.cpp

```diff
@@ -103,7 +103,7 @@ bool ModifierManager::add_modifier_effect(std::string_view identifier, bool posi
 		Logger::error("Invalid modifier effect identifier - empty!");
 		return false;
 	}
-	return modifier_effects.add_item({ identifier, positive_good, format });
+	return modifier_effects.add_item(std::unique_ptr<ModifierEffect> { new ModifierEffect { identifier, positive_good, format } });
```

Comment:
it is a bit cleaner to use `std::make_unique` for this

Reply:
I couldn't because `ModifierEffect`'s constructor is `private`, accessible by `ModifierManager` because it's a friend. Thinking about it again, I should be able to make `std::make_unique` a friend instead. That will publicly expose `ModifierEffect` construction though, at which point I might as well just make the constructor public and not need a friend declaration.

Reply:
Honestly making the constructor private makes little sense if the registry is locked, the member variables don't even have to be private (or const) necessarily if we disable the const functions of the registry, should however rename the non-const overload functions to something like `<name>_nc` or `<name>_mutable to clarify non-const/mutable return value.